README.md

```diff
@@ -535,6 +536,45 @@ func TenTimesReconciler() *reconcilers.SubReconciler[*buildv1alpha1.Function] {
 }
 ```
 
+#### ForEach
+
+A [`ForEach`](https://pkg.go.dev/reconciler.io/runtime/reconcilers#ForEach) calls the reconciler for each item returned from Items. The current cursor can be retrieved with [`CursorStasher`](https://pkg.go.dev/reconciler.io/runtime/reconcilers#CursorStasher).
+
+Nested iteration is allowed so long as the types being iterated over contain unique names. Otherwise the stash keys will collide. For testing the nested reconciler outside the scope of the loop, use the stasher's Key method to lookup the StashKey, do not expect the StashKey to be stable between releases.
```

Comment:
**[suggestion, non-blocking]:** When reading _"so long as the types being iterated over contain unique names"_ I was assuming that the items need to have unique names but it's speaking about types. I struggle to parse what it means for a "type" to "contain a name". The doc string of `ForEach` is a little clearer:

```go
// Multiple ForEach reconcilers are nestable so long as the types being iterated over are unique.
```

So the types must be unique then, I think. That makes sense, because we identify each nested reconciler by type.

Maybe this is better:

> Nested iteration is allowed so long as each reconciler's type is unique.

Reply:
Types have names in go. For example, the type `corev1.Pod` has the name `"Pod"`. Because the stash key is based on a string, we need to coerce the type to a string. In general this will be transparent to users, but if they have nested iteration of the same type or different types that have the same name, then a collision will occur.

I'll take a point to try to clean this up sooner rather than later.